### DISCUSSION TITLE
Hi, what is the minimum browser version alpine supports?

### DISCUSSION THREAD

**Discussion by helloyoucan** [alpinejs/alpine#2622] [1. Help] [answered]
If inscribe.

**Comment by danddanddand** [comment, marked as answer]
This is kind of a complicated question because you can get Alpine to work on most legacy browsers with a bit of effort by doing feature detection for required language feature and injecting the missing feature before alpine loads. 

For example, 
```
if ( typeof window.queueMicrotask !== "function" ) {
    window.queueMicrotask = function ( callback ) {
        Promise.resolve()
            .then( callback )
            .catch( ( e : any ) => setTimeout( () => { throw e; } ) );
    };
}
```

You can use [Core JS](https://github.com/zloirock/core-js) if you want to target widest range of browsers, but you'd want to do it lazily so you're not doing it on browsers that already support the feature you are targeting (forcing 100% of users to download tons of JS when <2% need it is bad)
